src/main/java/build/buf/protovalidate/internal/evaluator/MessageLike.java

```diff
@@ -0,0 +1,23 @@
+// Copyright 2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buf.protovalidate.internal.evaluator;
+
+import com.google.protobuf.Descriptors.FieldDescriptor;
+
+public interface MessageLike {
```

Comment:
How is this interface intended to be consumed by others? Should this be moved to another non-internal package?

Reply:
This is the critical interface for other implementations to implement, so yes, it should move to a non-internal package. I didn't pick one since that tends to be something maintainers want to do! I'll move it now and you all can bike-shed from there.